sql_generators/glean_usage/common.py

```diff
@@ -104,7 +104,11 @@ def table_names_from_baseline(baseline_table, include_project_id=True):
 def referenced_table_exists(view_sql):
     """Dry run the given view SQL to see if its referent exists."""
     dryrun = DryRun("foo/bar/view.sql", content=view_sql)
-    return 404 not in [e.get("code") for e in dryrun.errors()]
+    # 403 is returned if referenced dataset doesn't exist
+    # 404 is returned if referenced table or view doesn't exist
+    return 404 not in [e.get("code") for e in dryrun.errors()] and 403 not in [
+        e.get("code") for e in dryrun.errors()
+    ]
```

Comment:
rather than iterate over errors twice, this seems like a good place for `any`:
```suggestion
    return not any(e.get("code") in {403, 404} for e in dryrun.errors())
```